# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
tobs = pd.read_csv('hawaii_measurements.csv')
tobs.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
tobs.tail()

station        date  prcp  tobs
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

In [4]:
tobs.date.dtype

dtype('O')

In [5]:
# Convert the date column format from string to datetime
tobs.date = pd.to_datetime(tobs.date, infer_datetime_format=True)

In [6]:
# Set the date column as the DataFrame index
tobs = tobs.set_index(tobs["date"])
tobs.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [7]:
# Drop the date column
tobs = tobs.drop(columns = "date")

In [8]:
tobs.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [9]:
from scipy import stats

In [10]:
# "tobs" is "temperature observations"
df = pd.read_csv('hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [11]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [12]:
# Set the date column as the DataFrame index
df = df.set_index(df["date"])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [13]:
# Filter data for desired months
jun_df = df[df.index.month == 6]
dec_df = df[df.index.month == 12]

In [14]:
# Identify the average temperature for June
jun_df.mean()

<ipython-input-14-f5a0b91be0f8>:2: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  jun_df.mean()


prcp     0.136360
tobs    74.944118
dtype: float64

In [15]:
# Identify the average temperature for December
dec_df.mean()

<ipython-input-15-b792234d8235>:2: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  dec_df.mean()


prcp     0.216819
tobs    71.041529
dtype: float64

In [16]:
# Create collections of temperature data
jun_tobs = jun_df[['tobs']]
dec_tobs = dec_df[['tobs']]



In [17]:
# Run paired t-test
stats.ttest_ind(jun_tobs, dec_tobs)

Ttest_indResult(statistic=array([31.60372399]), pvalue=array([3.9025129e-191]))

### Analysis

#### The difference in the mean temperature between Jun and Dec in the year 2010-2017 is 3.9 degrees Celsius. The t-test was conducted and showed a p-value that is statistically significant.